src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -962,6 +984,19 @@ private void GeneratePropertyGetter<T>(QualifiedType returnType, T decl, Class @
                     ReturnType = decl.QualifiedType
                 };
 
+                var arrayType = field.Type as ArrayType;
+
+                if (arrayType != null && @class.IsValueType)
+                {
+                    CSharpTypePrinter typePrinter = new CSharpTypePrinter(ctx.Driver);
+                    string type = arrayType.CSharpType(typePrinter).Type;
+                    WriteLine(string.Format("fixed ({0} __arrPtr = {1}.{2})",
```

Comment:
I tries to push a Native context and then try. I also used VisitArrayType of typePrinter but nothing worked as required. Both the changes :
https://gist.github.com/genuinelucifer/079b00302845c486fb24
.
Both return 'fixed float' and required is 'float*'.... Now they don't even return []s....
Also, I tried CLI counterpart of typePrinter but it generates cli<> types which won't work...
.
If you could point me in the right direction, it would be quite helpful.